### ISSUE TITLE
Question: Is the network traffic of AllReduce(like, ML gradients) encrypted between workers?

### ISSUE THREAD

**Issue by jsyqrt** [kubeflow/mpi-operator#645]
Background
I've being trying to launch a MPI cluster to train a deep learning model with PyTorch's DDP. 

Question
Is the network traffic of AllReduce encrypted between workers?

**Comment by alculquicondor** [comment]
Actually, I'm not sure if all the communication is encrypted. But this is not a question for the operator. This is a question for the MPI implementation (OpenMPI, Intel, MPICH, etc).

/close

**Comment by jsyqrt** [comment]
Hi, I'am using the Intel MPI and it seems not support communication encryption. 

So I am curious that if it is possible to encrypt all MPI traffic with Istio/service mesh tools in KubeFlow?

**Comment by alculquicondor** [comment]
I think someone tried some time ago, but SSH didn't work well with it.

**Comment by rongou** [comment]
SSH is only used for the initial setup, the actual MPI traffic is not encrypted by design: https://stackoverflow.com/questions/6346873/how-do-mpi-implementations-openmpi-mpich-handle-security-authentication

**Comment by jsyqrt** [comment]
Istio says it will protect all traffic between pods, check [this](https://istio.io/latest/docs/concepts/security/).

Does this apply to KubeFlow and the MPI traffic in it? If so, maybe it's a good idea to use Istio together with KubeFlow

**Comment by tenzen-y** [comment]
The current mpi-operator specifications do not support istio as you can see here: https://github.com/kubeflow/mpi-operator/issues/480

**Comment by jsyqrt** [comment]
Hi @tenzen-y, thanks for your information about istio!
So is it possible to use a Service Mesh with mpi-operator, to protect mpi ccl traffics?

**Comment by alculquicondor** [comment]
It might be possible, but nobody has reported doing it so far.
